org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ConnectionStreamFactory.java

```diff
@@ -110,6 +264,10 @@ private StreamProvider createProvider() {
 		if (port != null) {
 			return new SocketStreamProvider(JDTEnvironmentUtils.getClientHost(), port);
 		}
+		File pipeFile = getPipeFile();
```

Comment:
We should probably include a comment that calling `exists()` or accessing file metadata when this respresents a named pipe (eg. windows) will close the underlying file causing failure.